extensions/big-data-cluster/src/bigDataCluster/dialog/addControllerDialog.ts

```diff
@@ -0,0 +1,494 @@
+				}
+			});
+
+			wrapperContainer.addItem(usernameInputContainer, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'margin': '-2px 0px 0px 0px',
+					'width': '100%'
+				}
+			});
+
+			let passwordInputContainer: azdata.FlexContainer = this.uiModelBuilder.flexContainer()
+				.withLayout(<azdata.FlexLayout>{
+					flexFlow: 'row',
+					alignItems: 'center',
+					width: '100%'
+				})
+				.component();
+
+			let passwordInputText = this.uiModelBuilder.text()
+				.withProperties<azdata.TextComponentProperties>({ value: 'Password' }).component();
+
+			this.passwordInputBox = this.uiModelBuilder.inputBox()
+				.withProperties<azdata.InputBoxProperties>({
+					placeHolder: 'password',
+					inputType: 'password',
+					value: this.model.prefilledPassword
+				})
+				.component();
+			this.passwordInputBox.onTextChanged(e => this.toggleSignInButton());
+
+			passwordInputContainer.addItem(passwordInputText, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'width': '80px'
+				}
+			});
+
+			passwordInputContainer.addItem(this.passwordInputBox, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'margin': '-2px 0px 0px 0px',
+					'width': '100%',
+				}
+			});
+
+			wrapperContainer.addItem(passwordInputContainer, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'margin': '-2px 0px 0px 0px',
+					'width': '100%'
+				}
+			});
+
+			this.rememberPwCheckBox = this.uiModelBuilder.checkBox()
+				.withProperties<azdata.CheckBoxProperties>({
+					label: localize('bigDataClusters.addControllerDialog.rememberPassword', 'Remember Password'),
+					checked: this.model.prefilledRememberPassword
+				}).component();
+			this.rememberPwCheckBox.enabled = false;
+
+			wrapperContainer.addItem(this.rememberPwCheckBox, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'margin': '0px 0px 0px 65px',
+				}
+			});
+
+			this.errorTextContainer = this.uiModelBuilder.flexContainer()
+				.withLayout(<azdata.FlexLayout>{
+					flexFlow: 'row',
+					alignItems: 'center',
+					width: '100%'
+				})
+				.component();
+
+			wrapperContainer.addItem(this.errorTextContainer, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					width: '100%'
+				}
+			});
+
+			titledContainer.addContentContainer(wrapperContainer);
+			await view.initializeModel(titledContainer.flexContainer);
+		});
+
+		this.dialog.registerCloseValidator(async () => await this.validate());
+		this.dialog.cancelButton.onClick(async () => await this.cancel());
+		this.dialog.okButton.label = this.SignInButtonText;
+		this.dialog.cancelButton.label = this.CancelButtonText;
+	}
+
+	private toggleErrorText(errorMessage?: string): void {
```

Comment:
>toggleErrorText [](start = 9, length = 15)

This isn't really toggling anything - it's more setErrorText

Reply:
it toggles. Will be replaced with built in dialog error message box. I will leave it as it is for now.

Reply:
I guess I'll wait and see your changes - but toggling something means that calling the method (typically which doesn't have any parameters) switches between 'on' and 'off' states. If I call this method as it is with an empty string it doesn't change anything - it'll always continue to just clear the messages. So as this code is now the name is incorrect.

Reply:
As I mentioned the entire method will be removed, and built in error box in dialog will be used instead.

Reply:
Are you planning on doing this in this PR? Because there's no reason you should check in code that's misleading - even if it's only temporary. 

---
In reply to: [299702909](https://github.com/microsoft/azuredatastudio/pull/6204#discussion_r299702909) [](ancestors = 299702909)